### PR TITLE
Update pytest command flags

### LINKED ISSUE #22855: [Key Vault] Use pytest command flags for test run config
For example, refer to [Identity's use of pytest command flags](https://github.com/Azure/azure-sdk-for-python/blob/d9f9a4e04d7807834fcde51364ec2265a02a1af6/sdk/identity/azure-identity/conftest.py#L49) to run or skip certain tests. By allowing users to pass an argument to `pytest`, like `--all-api-versions`, we could override [this check](https://github.com/Azure/azure-sdk-for-python/blob/de91b99a98f3d6327c2436b6012aebfea9ab18e7/sdk/keyvault/azure-keyvault-keys/tests/_test_case.py#L106) that skips non-default API versions in live runs instead of needing to comment this check out when recording for all API versions.

### PR DESCRIPTION
**Pull request by lbhsos**
Resolves #22855
Use pytest command flags(--all-api-versions) for test run config

If provide the `--all-api-versions` option at the starting the test, will test all api versions
These belows are examples:
```bash
pytest test_crypto_client.py -v --all-api-versions
===================================================== test session starts =====================================================
platform darwin -- Python 3.8.2, pytest-7.1.2, pluggy-1.0.0 -- /Users/bohyun/Desktop/Work/python/.venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/bohyun/Desktop/Work/python/test/azure-sdk-for-python, configfile: setup.cfg
plugins: asyncio-0.19.0, cov-3.0.0
asyncio: mode=strict
collected 120 items

test_crypto_client.py::TestCryptoClient::test_ec_key_id[7.3_mhsm] PASSED                                                [  0%]
test_crypto_client.py::TestCryptoClient::test_ec_key_id[7.3_vault] PASSED                                               [  1%]
test_crypto_client.py::TestCryptoClient::test_ec_key_id[7.2_mhsm] PASSED                                                [  2%]
test_crypto_client.py::TestCryptoClient::test_ec_key_id[7.2_vault] PASSED                                               [  3%]
test_crypto_client.py::TestCryptoClient::test_ec_key_id[7.1_vault] PASSED                                               [  4%]
test_crypto_client.py::TestCryptoClient::test_ec_key_id[7.0_vault] PASSED                                               [  5%]
test_crypto_client.py::TestCryptoClient::test_ec_key_id[2016-10-01_vault] PASSED                                        [  5%]
test_crypto_client.py::TestCryptoClient::test_rsa_key_id[7.3_mhsm] PASSED                                               [  6%]
test_crypto_client.py::TestCryptoClient::test_rsa_key_id[7.3_vault] PASSED                                              [  7%]
test_crypto_client.py::TestCryptoClient::test_rsa_key_id[7.2_mhsm] PASSED                                               [  8%]
test_crypto_client.py::TestCryptoClient::test_rsa_key_id[7.2_vault] PASSED                                              [  9%]
test_crypto_client.py::TestCryptoClient::test_rsa_key_id[7.1_vault] PASSED                                              [ 10%]
test_crypto_client.py::TestCryptoClient::test_rsa_key_id[7.0_vault] PASSED                                              [ 10%]
test_crypto_client.py::TestCryptoClient::test_rsa_key_id[2016-10-01_vault] PASSED                                       [ 11%]
test_crypto_client.py::TestCryptoClient::test_encrypt_and_decrypt[7.3_mhsm] PASSED                                      [ 12%]
test_crypto_client.py::TestCryptoClient::test_encrypt_and_decrypt[7.3_vault] PASSED                                     [ 13%]
test_crypto_client.py::TestCryptoClient::test_encrypt_and_decrypt[7.2_mhsm] PASSED                                      [ 14%]
test_crypto_client.py::TestCryptoClient::test_encrypt_and_decrypt[7.2_vault] PASSED                                     [ 15%]
test_crypto_client.py::TestCryptoClient::test_encrypt_and_decrypt[7.1_vault] PASSED                                     [ 15%]
test_crypto_client.py::TestCryptoClient::test_encrypt_and_decrypt[7.0_vault] PASSED                                     [ 16%]
test_crypto_client.py::TestCryptoClient::test_encrypt_and_decrypt[2016-10-01_vault] PASSED                              [ 17%]
test_crypto_client.py::TestCryptoClient::test_sign_and_verify[7.3_mhsm] PASSED                                          [ 18%]
test_crypto_client.py::TestCryptoClient::test_sign_and_verify[7.3_vault] PASSED                                         [ 19%]
test_crypto_client.py::TestCryptoClient::test_sign_and_verify[7.2_mhsm] PASSED                                          [ 20%]
test_crypto_client.py::TestCryptoClient::test_sign_and_verify[7.2_vault] PASSED                                         [ 20%]
test_crypto_client.py::TestCryptoClient::test_sign_and_verify[7.1_vault] PASSED                                         [ 21%]
test_crypto_client.py::TestCryptoClient::test_sign_and_verify[7.0_vault] PASSED                                         [ 22%]
test_crypto_client.py::TestCryptoClient::test_sign_and_verify[2016-10-01_vault] PASSED                                  [ 23%]
test_crypto_client.py::TestCryptoClient::test_wrap_and_unwrap[7.3_mhsm] PASSED                                          [ 24%]
test_crypto_client.py::TestCryptoClient::test_wrap_and_unwrap[7.3_vault] PASSED                                         [ 25%]
test_crypto_client.py::TestCryptoClient::test_wrap_and_unwrap[7.2_mhsm] PASSED                                          [ 25%]
test_crypto_client.py::TestCryptoClient::test_wrap_and_unwrap[7.2_vault] PASSED                                         [ 26%]
test_crypto_client.py::TestCryptoClient::test_wrap_and_unwrap[7.1_vault] PASSED                                         [ 27%]
test_crypto_client.py::TestCryptoClient::test_wrap_and_unwrap[7.0_vault] PASSED                                         [ 28%]
test_crypto_client.py::TestCryptoClient::test_wrap_and_unwrap[2016-10-01_vault] PASSED                                  [ 29%]
test_crypto_client.py::TestCryptoClient::test_symmetric_encrypt_and_decrypt[7.3_mhsm] PASSED                            [ 30%]
test_crypto_client.py::TestCryptoClient::test_symmetric_encrypt_and_decrypt[7.2_mhsm] PASSED                            [ 30%]
test_crypto_client.py::TestCryptoClient::test_symmetric_wrap_and_unwrap[7.3_mhsm] PASSED                                [ 31%]
test_crypto_client.py::TestCryptoClient::test_symmetric_wrap_and_unwrap[7.2_mhsm] PASSED                                [ 32%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local[7.3_mhsm] PASSED                                            [ 33%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local[7.3_vault] PASSED                                           [ 34%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local[7.2_mhsm] PASSED                                            [ 35%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local[7.2_vault] PASSED                                           [ 35%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local[7.1_vault] PASSED                                           [ 36%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local[7.0_vault] PASSED                                           [ 37%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local[2016-10-01_vault] PASSED                                    [ 38%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local_from_jwk[7.3_mhsm] PASSED                                   [ 39%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local_from_jwk[7.3_vault] PASSED                                  [ 40%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local_from_jwk[7.2_mhsm] PASSED                                   [ 40%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local_from_jwk[7.2_vault] PASSED                                  [ 41%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local_from_jwk[7.1_vault] PASSED                                  [ 42%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local_from_jwk[7.0_vault] PASSED                                  [ 43%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local_from_jwk[2016-10-01_vault] PASSED                           [ 44%]
test_crypto_client.py::TestCryptoClient::test_symmetric_encrypt_local[7.3_mhsm] PASSED                                  [ 45%]
test_crypto_client.py::TestCryptoClient::test_symmetric_encrypt_local[7.2_mhsm] PASSED                                  [ 45%]
test_crypto_client.py::TestCryptoClient::test_symmetric_decrypt_local[7.3_mhsm] PASSED                                  [ 46%]
test_crypto_client.py::TestCryptoClient::test_symmetric_decrypt_local[7.2_mhsm] PASSED                                  [ 47%]
test_crypto_client.py::TestCryptoClient::test_wrap_local[7.3_mhsm] PASSED                                               [ 48%]
test_crypto_client.py::TestCryptoClient::test_wrap_local[7.3_vault] PASSED                                              [ 49%]
test_crypto_client.py::TestCryptoClient::test_wrap_local[7.2_mhsm] PASSED                                               [ 50%]
test_crypto_client.py::TestCryptoClient::test_wrap_local[7.2_vault] PASSED                                              [ 50%]
test_crypto_client.py::TestCryptoClient::test_wrap_local[7.1_vault] PASSED                                              [ 51%]
test_crypto_client.py::TestCryptoClient::test_wrap_local[7.0_vault] PASSED                                              [ 52%]
test_crypto_client.py::TestCryptoClient::test_wrap_local[2016-10-01_vault] PASSED                                       [ 53%]
test_crypto_client.py::TestCryptoClient::test_wrap_local_from_jwk[7.3_mhsm] PASSED                                      [ 54%]
test_crypto_client.py::TestCryptoClient::test_wrap_local_from_jwk[7.3_vault] PASSED                                     [ 55%]
test_crypto_client.py::TestCryptoClient::test_wrap_local_from_jwk[7.2_mhsm] PASSED                                      [ 55%]
test_crypto_client.py::TestCryptoClient::test_wrap_local_from_jwk[7.2_vault] PASSED                                     [ 56%]
test_crypto_client.py::TestCryptoClient::test_wrap_local_from_jwk[7.1_vault] PASSED                                     [ 57%]
test_crypto_client.py::TestCryptoClient::test_wrap_local_from_jwk[7.0_vault] PASSED                                     [ 58%]
test_crypto_client.py::TestCryptoClient::test_wrap_local_from_jwk[2016-10-01_vault] PASSED                              [ 59%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local[7.3_mhsm] PASSED                                         [ 60%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local[7.3_vault] PASSED                                        [ 60%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local[7.2_mhsm] PASSED                                         [ 61%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local[7.2_vault] PASSED                                        [ 62%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local[7.1_vault] PASSED                                        [ 63%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local[7.0_vault] PASSED                                        [ 64%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local[2016-10-01_vault] PASSED                                 [ 65%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local_from_jwk[7.3_mhsm] PASSED                                [ 65%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local_from_jwk[7.3_vault] PASSED                               [ 66%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local_from_jwk[7.2_mhsm] PASSED                                [ 67%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local_from_jwk[7.2_vault] PASSED                               [ 68%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local_from_jwk[7.1_vault] PASSED                               [ 69%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local_from_jwk[7.0_vault] PASSED                               [ 70%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local_from_jwk[2016-10-01_vault] PASSED                        [ 70%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local[7.3_mhsm] PASSED                                          [ 71%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local[7.3_vault] PASSED                                         [ 72%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local[7.2_mhsm] PASSED                                          [ 73%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local[7.2_vault] PASSED                                         [ 74%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local[7.1_vault] PASSED                                         [ 75%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local[7.0_vault] PASSED                                         [ 75%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local[2016-10-01_vault] PASSED                                  [ 76%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local_from_jwk[7.3_mhsm] PASSED                                 [ 77%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local_from_jwk[7.3_vault] PASSED                                [ 78%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local_from_jwk[7.2_mhsm] PASSED                                 [ 79%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local_from_jwk[7.2_vault] PASSED                                [ 80%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local_from_jwk[7.1_vault] PASSED                                [ 80%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local_from_jwk[7.0_vault] PASSED                                [ 81%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local_from_jwk[2016-10-01_vault] PASSED                         [ 82%]
test_crypto_client.py::TestCryptoClient::test_local_validity_period_enforcement[7.3_mhsm] PASSED                        [ 83%]
test_crypto_client.py::TestCryptoClient::test_local_validity_period_enforcement[7.3_vault] PASSED                       [ 84%]
test_crypto_client.py::TestCryptoClient::test_local_validity_period_enforcement[7.2_mhsm] PASSED                        [ 85%]
test_crypto_client.py::TestCryptoClient::test_local_validity_period_enforcement[7.2_vault] PASSED                       [ 85%]
test_crypto_client.py::TestCryptoClient::test_local_validity_period_enforcement[7.1_vault] PASSED                       [ 86%]
test_crypto_client.py::TestCryptoClient::test_local_validity_period_enforcement[7.0_vault] PASSED                       [ 87%]
test_crypto_client.py::TestCryptoClient::test_local_validity_period_enforcement[2016-10-01_vault] PASSED                [ 88%]
test_crypto_client.py::test_custom_hook_policy PASSED                                                                   [ 89%]
test_crypto_client.py::test_initialization_given_key PASSED                                                             [ 90%]
test_crypto_client.py::test_initialization_get_key_successful PASSED                                                    [ 90%]
test_crypto_client.py::test_initialization_forbidden_to_get_key PASSED                                                  [ 91%]
test_crypto_client.py::test_initialization_transient_failure_getting_key PASSED                                         [ 92%]
test_crypto_client.py::test_calls_service_for_operations_unsupported_locally PASSED                                     [ 93%]
test_crypto_client.py::test_local_only_mode_no_service_calls PASSED                                                     [ 94%]
test_crypto_client.py::test_local_only_mode_raise PASSED                                                                [ 95%]
test_crypto_client.py::test_prefers_local_provider PASSED                                                               [ 95%]
test_crypto_client.py::test_aes_cbc_key_size_validation PASSED                                                          [ 96%]
test_crypto_client.py::test_aes_cbc_iv_validation PASSED                                                                [ 97%]
test_crypto_client.py::test_encrypt_argument_validation PASSED                                                          [ 98%]
test_crypto_client.py::test_decrypt_argument_validation PASSED                                                          [ 99%]
test_crypto_client.py::test_retain_url_port PASSED                                                                      [100%]

==================================================== 120 passed in 16.30s =====================================================

```

and without `--all-api-versions`, will test only default api versions
```
pytest test_crypto_client.py -v
===================================================== test session starts =====================================================
platform darwin -- Python 3.8.2, pytest-7.1.2, pluggy-1.0.0 -- /Users/bohyun/Desktop/Work/python/.venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/bohyun/Desktop/Work/python/test/azure-sdk-for-python, configfile: setup.cfg
plugins: asyncio-0.19.0, cov-3.0.0
asyncio: mode=strict
collected 46 items

test_crypto_client.py::TestCryptoClient::test_ec_key_id[7.3_mhsm] PASSED                                                [  2%]
test_crypto_client.py::TestCryptoClient::test_ec_key_id[7.3_vault] PASSED                                               [  4%]
test_crypto_client.py::TestCryptoClient::test_rsa_key_id[7.3_mhsm] PASSED                                               [  6%]
test_crypto_client.py::TestCryptoClient::test_rsa_key_id[7.3_vault] PASSED                                              [  8%]
test_crypto_client.py::TestCryptoClient::test_encrypt_and_decrypt[7.3_mhsm] PASSED                                      [ 10%]
test_crypto_client.py::TestCryptoClient::test_encrypt_and_decrypt[7.3_vault] PASSED                                     [ 13%]
test_crypto_client.py::TestCryptoClient::test_sign_and_verify[7.3_mhsm] PASSED                                          [ 15%]
test_crypto_client.py::TestCryptoClient::test_sign_and_verify[7.3_vault] PASSED                                         [ 17%]
test_crypto_client.py::TestCryptoClient::test_wrap_and_unwrap[7.3_mhsm] PASSED                                          [ 19%]
test_crypto_client.py::TestCryptoClient::test_wrap_and_unwrap[7.3_vault] PASSED                                         [ 21%]
test_crypto_client.py::TestCryptoClient::test_symmetric_encrypt_and_decrypt[7.3_mhsm] PASSED                            [ 23%]
test_crypto_client.py::TestCryptoClient::test_symmetric_wrap_and_unwrap[7.3_mhsm] PASSED                                [ 26%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local[7.3_mhsm] PASSED                                            [ 28%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local[7.3_vault] PASSED                                           [ 30%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local_from_jwk[7.3_mhsm] PASSED                                   [ 32%]
test_crypto_client.py::TestCryptoClient::test_encrypt_local_from_jwk[7.3_vault] PASSED                                  [ 34%]
test_crypto_client.py::TestCryptoClient::test_symmetric_encrypt_local[7.3_mhsm] PASSED                                  [ 36%]
test_crypto_client.py::TestCryptoClient::test_symmetric_decrypt_local[7.3_mhsm] PASSED                                  [ 39%]
test_crypto_client.py::TestCryptoClient::test_wrap_local[7.3_mhsm] PASSED                                               [ 41%]
test_crypto_client.py::TestCryptoClient::test_wrap_local[7.3_vault] PASSED                                              [ 43%]
test_crypto_client.py::TestCryptoClient::test_wrap_local_from_jwk[7.3_mhsm] PASSED                                      [ 45%]
test_crypto_client.py::TestCryptoClient::test_wrap_local_from_jwk[7.3_vault] PASSED                                     [ 47%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local[7.3_mhsm] PASSED                                         [ 50%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local[7.3_vault] PASSED                                        [ 52%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local_from_jwk[7.3_mhsm] PASSED                                [ 54%]
test_crypto_client.py::TestCryptoClient::test_rsa_verify_local_from_jwk[7.3_vault] PASSED                               [ 56%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local[7.3_mhsm] PASSED                                          [ 58%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local[7.3_vault] PASSED                                         [ 60%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local_from_jwk[7.3_mhsm] PASSED                                 [ 63%]
test_crypto_client.py::TestCryptoClient::test_ec_verify_local_from_jwk[7.3_vault] PASSED                                [ 65%]
test_crypto_client.py::TestCryptoClient::test_local_validity_period_enforcement[7.3_mhsm] PASSED                        [ 67%]
test_crypto_client.py::TestCryptoClient::test_local_validity_period_enforcement[7.3_vault] PASSED                       [ 69%]
test_crypto_client.py::test_custom_hook_policy PASSED                                                                   [ 71%]
test_crypto_client.py::test_initialization_given_key PASSED                                                             [ 73%]
test_crypto_client.py::test_initialization_get_key_successful PASSED                                                    [ 76%]
test_crypto_client.py::test_initialization_forbidden_to_get_key PASSED                                                  [ 78%]
test_crypto_client.py::test_initialization_transient_failure_getting_key PASSED                                         [ 80%]
test_crypto_client.py::test_calls_service_for_operations_unsupported_locally PASSED                                     [ 82%]
test_crypto_client.py::test_local_only_mode_no_service_calls PASSED                                                     [ 84%]
test_crypto_client.py::test_local_only_mode_raise PASSED                                                                [ 86%]
test_crypto_client.py::test_prefers_local_provider PASSED                                                               [ 89%]
test_crypto_client.py::test_aes_cbc_key_size_validation PASSED                                                          [ 91%]
test_crypto_client.py::test_aes_cbc_iv_validation PASSED                                                                [ 93%]
test_crypto_client.py::test_encrypt_argument_validation PASSED                                                          [ 95%]
test_crypto_client.py::test_decrypt_argument_validation PASSED                                                          [ 97%]
test_crypto_client.py::test_retain_url_port PASSED                                                                      [100%]

===================================================== 46 passed in 6.24s ======================================================
```

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by mccoyp — COMMENTED** [review]
Hi @ponopono0322, thank you for opening a PR! This is really great, and seems to get us 90% of the way there 🙂

My tracking issue wasn't specific enough though, so there's a slight change we need to make. The line I referenced in https://github.com/Azure/azure-sdk-for-python/issues/22855 didn't have a commit ID so the reference was lost before, but [this](https://github.com/Azure/azure-sdk-for-python/blob/de91b99a98f3d6327c2436b6012aebfea9ab18e7/sdk/keyvault/azure-keyvault-keys/tests/_test_case.py#L106) is the actual condition we were thinking of overriding.

In other words, we still want to run tests for all API versions by default in playback mode. And, we still want to only run the default API version by default in live mode. But when we provide the `--all-api-versions` option in live mode, then we want to run tests for all API versions.

Would you be able to modify this to apply that change? Please let me know if you have any questions or feedback!

**Comment by ponopono0322** [comment]
Hello @mccoyp, thank you for leaving a comment.
Ask to confirm that the code is consistent with the intention of the change.

When I checked, the live mode seems to be executed through the `AZURE_TEST_RUN_LIVE` environment variable.

Therefore, if `AZURE_TEST_RUN_LIVE` and `--all-api-versions` are `true` or `yes`, set `all_api_version`.
Else `default_version`.

So, legacy code:
```python
def pytest_configure(config):
    if config.getoption(ALL_API_VERSIONS) == 'false':
        pytest.api_version = [DEFAULT_VERSION]
    else:
        pytest.api_version = ApiVersion
```

To be:
```python
def pytest_configure(config):
    get_live_env = os.environ.get("AZURE_TEST_RUN_LIVE")
    if get_live_env:
        get_live_env = str(get_live_env).lower()

    if get_live_env in ['yes', 'true'] and config.getoption(ALL_API_VERSIONS):
        pytest.api_version = [DEFAULT_VERSION]
    else:
        pytest.api_version = ApiVersion
```

**Comment by mccoyp** [comment]
Hi @ponopono0322, that's correct! And we can simplify the code even further by using the `is_live` helper method that we already import from `devtools_testutils` in KV's `conftest.py`:
```python
def pytest_addoption(parser):
    parser.addoption(ALL_API_VERSIONS, action="store_true", default=False,
                     help="Test all api version in live mode. Not applicable in playback mode.")


def pytest_configure(config):
    if is_live() and not config.getoption(ALL_API_VERSIONS):
        pytest.api_version = [DEFAULT_VERSION]
    else:
        pytest.api_version = ApiVersion
```
i.e. if we're running live tests and `--all-api-versions` isn't provided, only use the default version. Otherwise, use all API versions.

I should be able to push this change to the PR if you'd like, which may be easiest since the PR has been up for a while. I don't want to take over your contributions though, so I'd be happy to let you make the changes instead 🙂 Just let me know!

**Comment by ponopono0322** [comment]
Oh.. I made a mistake while working on the branch. I will close the current PR. I'm sorry.
